### ISSUE TITLE
Cinnamon displays all Java application icons in top left corner when minimized/hidden to tray

### ISSUE THREAD

**Issue by nick-s-b** [linuxmint/cinnamon#11600]
```
 * Cinnamon version (cinnamon --version)
Cinnamon 5.6.8
 * Distribution - (Mint 21, Arch, Fedora 37, etc...)
 Arch
 * Graphics hardware *and* driver used
 NVIDIA-SMI 530.41.03              Driver Version: 530.41.03    CUDA Version: 12.1  
 * 32 or 64 bit
 *64-bit
 * Attach ~/.xsession-errors, or /var/log/syslog
![xsession-errors.zip](https://github.com/linuxmint/cinnamon/files/11116419/xsession-errors.zip)
 
 ```

### Issue
If you start and then minimize to system tray any Java-based application, the app's icon will actually show up on Cinnamon's desktop in the top left corner of the screen.

### Steps to reproduce
Install and run one of the multitude of Java applications (say, jDownloader) and minimize it to system tray by clicking the close button and selecting "Hide to Tray" option.
![Screenshot at 2023-03-30 17-50-22-fs8](https://user-images.githubusercontent.com/7957590/228972317-f19af258-6dd9-4164-9b13-e8816fc6d099.png)
Notice that once you do so, program's icon will show up in top left corner of the screen.
![Screenshot at 2023-03-30 17-41-35-fs8](https://user-images.githubusercontent.com/7957590/228972083-22d49526-dcbb-4014-b8dc-93bfe7478abd.png)

### Expected behaviour
There should be no icon on the desktop.

### Other information
This happens with every other Java-based application I've tried (including IntelliJ).

**Comment by arunmandal53** [comment]
Same Issue With GloboNote.

**Comment by Tralen** [comment]
Hello,

I'm having the same exact issue with PJe (brazilian judiciary digital certificate token). It is a real problem because the only way to authenticate the token is by right clicking the icon at the system tray, which I'm unable to do.

Mint 21.1 - Cinnamon 5.6.8

**Comment by romario74** [comment]
Same here with jDownloader and MediathekView.

**Comment by ugackMiner53** [comment]
I'm having this issue with a program that I wrote myself using Java AWT's `SystemTray` in a Swing based app. The icon appears in the system tray, but completely black. Interacting with it works, but the icon is missing, and only appears in the desktop on the top left.

The application icon works correctly outside of the system tray.
![image](https://github.com/linuxmint/cinnamon/assets/55325485/6cdb6f21-8deb-4d82-afed-f579e59260f5)
![image](https://github.com/linuxmint/cinnamon/assets/55325485/eacbe3c8-61f3-426c-b4f0-2f9eb9599c19)

**Comment by aless2003** [comment]
Can confirm, functionality is there, but the visuals bug out in the form of a missing icon in the system tray itself with the icon being rendered on the desktop instead

**Comment by skels130** [comment]
Distribution
Mint 21.3 and Ubuntu Cinnamon 24.04 LTS

Package version
6.0.4

Graphics hardware in use
Intel Iris XE

Frequency
Always

Bug description
The tray icon for Connectwise Screenconnect displays in the upper right corner rather than in the system tray, where a black box is displayed. Icon is sometimes rendered as a gray box instead of the icon as well.
ubuntu 24.04 (vm)
![image](https://github.com/linuxmint/cinnamon/assets/13512845/c6169f8d-e641-4bae-9801-6663c4e86b02)
Mint 21.3 (installed)
![image](https://github.com/linuxmint/cinnamon/assets/13512845/eed9c2c8-03f9-407d-b351-050babb2331a)

Steps to reproduce
Install the Connectwise Screenconnect/control application, and it displays this behavior on all tested distributions with Cinnamon, leading me to believe this is a Cinnamon bug. Co-worker running Xfce version of Mint does not see this behavior.

**Comment by Philippe734** [comment]
Another example with Universal Media Server, a Java app.

Linux Mint 21.3
Mate VS Cinnamon
Mate : OK
Cinnamon : not OK

![3TZfDO3](https://github.com/linuxmint/cinnamon/assets/24923693/e9447047-ade4-4f91-a7f8-85f4862dc52a)
![rLampBS](https://github.com/linuxmint/cinnamon/assets/24923693/46ae4722-bda7-4155-958d-224f74ba1f61)

**Comment by switzel** [comment]
This is over a year old and presumably something very trivial for someone who knows where to look. Isn't there a cinnamon insider who could give is a clue?

**Comment by Tralen** [comment]
Yep, just did a fresh install on two computers and the issue still persists. Mint 21.3, trying to run PJe and Shodo, two programs for brazilian judiciary certificate tokens.

**Comment by xGSTx** [comment]
Issue still present with JDownloader and Mint 22 / Cinnamon 6.2.9. For me it is not even possible to interact with the black icon in the System Tray, It is simply there and clicking on it has no effect. This issue is over 2 Years old now.... 

Please fix this!

**Comment by ShotSwine** [comment]
This issue is still present as of December 2024...

**Comment by hayden-servaenergy** [comment]
Can confirm this is still an issue in Cinnamon 6.4.8

**Comment by envyniv** [comment]
Still an issue in the latest version of Cinnamon.
